contracts/registration/RegistrationSimple.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+import {ICertificateDispatcher} from "../interfaces/dispatchers/ICertificateDispatcher.sol";
+
+import {IPassportDispatcher} from "../interfaces/dispatchers/IPassportDispatcher.sol";
+
+import {MultiOwnable} from "@solarity/solidity-lib/access/MultiOwnable.sol";
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {PoseidonSMT} from "../state/PoseidonSMT.sol";
+
+import {StateKeeper} from "../state/StateKeeper.sol";
+import {TSSUpgradeable} from "../state/TSSUpgradeable.sol";
+import {VerifierHelper} from "@solarity/solidity-lib/libs/zkp/snarkjs/VerifierHelper.sol";
+
+contract RegistrationSimple is Initializable, MultiOwnable, TSSUpgradeable {
+    using ECDSA for bytes32;
+    using MerkleProof for bytes32[];
+    using VerifierHelper for address;
+
+    string public constant REGISTRATION_SIMPLE_PREFIX = "Registration simple prefix";
+
+    uint256 internal constant _PROOF_SIGNALS_COUNT = 3;
+
+    struct Passport {
+        uint256 dgCommit;
+        bytes32 dg1Hash;
+        bytes32 publicKey;
+        bytes32 passportHash;
+        address verifier;
+    }
+
+    StateKeeper public stateKeeper;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function __RegistrationSimple_init(
+        address tssSigner_,
+        string calldata chainName_,
+        address stateKeeper_
+    ) external initializer {
+        __MultiOwnable_init();
+        __TSSSigner_init(tssSigner_, chainName_);
+
+        stateKeeper = StateKeeper(stateKeeper_);
+    }
+
+    function registerSimple(
+        uint256 identityKey_,
+        Passport memory passport_,
+        bytes memory signature_,
```

Comment:
We need to make this signature invalid the next time. Use `stateKeeper.useSignature()` method. It is important to keep the `registration` stateless (config is fine).

Reply:
ECDSA signatures are generally not unique and using them as nonce is not cool, however openzeppelin ECDSA lib enforces that.

Reply:
Added the `stateKeeper.useSignature(keccak256(signature_))`, not sure if it is enough